repo/laghos/package.py

```diff
@@ -0,0 +1,82 @@
+
+from spack.package import *
+
+
+class Laghos(MakefilePackage):
+    """Laghos (LAGrangian High-Order Solver) is a CEED miniapp that solves the
+    time-dependent Euler equations of compressible gas dynamics in a moving
+    Lagrangian frame using unstructured high-order finite element spatial
+    discretization and explicit high-order time-stepping.
+    """
+
+    tags = ["proxy-app", "ecp-proxy-app"]
+
+    homepage = "https://computing.llnl.gov/projects/co-design/laghos"
+    url = "https://github.com/CEED/Laghos/archive/v1.0.tar.gz"
+    git = "https://github.com/CEED/Laghos.git"
+
+    maintainers("v-dobrev", "tzanio", "vladotomov")
+
+    license("BSD-2-Clause")
+
+    version("develop", branch="master")
+    version("3.1", sha256="49b65edcbf9732c7f6c228958620e18980c43ad8381315a8ba9957ecb7534cd5")
+    version("3.0", sha256="4db56286e15b42ecdc8d540c4888a7dec698b019df9c7ccb8319b7ea1f92d8b4")
+    version("2.0", sha256="dd3632d5558889beec2cd3c49eb60f633f99e6d886ac868731610dd006c44c14")
+    version("1.1", sha256="53b9bfe2af263c63eb4544ca1731dd26f40b73a0d2775a9883db51821bf23b7f")
+    version("1.0", sha256="af50a126355a41c758fcda335a43fdb0a3cd97e608ba51c485afda3dd84a5b34")
+
+    variant("metis", default=True, description="Enable/disable METIS support")
+    variant("ofast", default=False, description="Enable gcc optimization flags")
+
+    depends_on("mfem+mpi+metis", when="+metis")
+    depends_on("mfem+mpi~metis", when="~metis")
+
+    depends_on("mfem@develop", when="@develop")
+    depends_on("mfem@4.2.0:", when="@3.1")
+    depends_on("mfem@4.1.0:4.1", when="@3.0")
+    # Recommended mfem version for laghos v2.0 is: ^mfem@3.4.1-laghos-v2.0
+    depends_on("mfem@3.4.1-laghos-v2.0", when="@2.0")
+    # Recommended mfem version for laghos v1.x is: ^mfem@3.3.1-laghos-v1.0
+    depends_on("mfem@3.3.1-laghos-v1.0", when="@1.0,1.1")
+
+    # Replace MPI_Session
+    patch(
+        "https://github.com/CEED/Laghos/commit/c800883ab2741c8c3b99486e7d8ddd8e53a7cb95.patch?full_index=1",
+        sha256="e783a71c3cb36886eb539c0f7ac622883ed5caf7ccae597d545d48eaf051d15d",
+        when="@3.1 ^mfem@4.4:",
+    )
+
+    @property
+    def build_targets(self):
+        targets = []
+        spec = self.spec
+
+        targets.append("MFEM_DIR=%s" % spec["mfem"].prefix)
+        targets.append("CONFIG_MK=%s" % spec["mfem"].package.config_mk)
+        targets.append("TEST_MK=%s" % spec["mfem"].package.test_mk)
+        if spec.satisfies("@:2.0"):
+            targets.append("CXX=%s" % spec["mpi"].mpicxx)
+        if "+ofast %gcc" in self.spec:
+            targets.append("CXXFLAGS = -Ofast -finline-functions")
+        return targets
+
+    # See lib/spack/spack/build_systems/makefile.py
+    def check(self):
+        with working_dir(self.build_directory):
+            make("test", *self.build_targets)
+ 
+    def install(self, spec, prefix):
+        mkdirp(prefix.bin)
+        install("laghos", prefix.bin)
+
+    @run_after("install")
```

Comment:
@scheibelp Is run_after("install") the correct way to copy data files after installing the binary?

Reply:
This should work fine. Is there a failure at this step?

Using `cache_extra_test_sources` assumes the files live in `.../stage/spack-src/data` and that you want them in `<install-prefix>/data`

Reply:
I tested this on a fresh `benchpark` install.  It looks to me like the bit of code that I hoped would  copy the `data` directory to `install-prefix/data` actually copied it to `install-prefix/.spack/test/data`.

Reply:
Whoops sorry yes: you'll need to use `install_tree` then (which allows setting up a `src` and a `dst`):

```
install_tree("data", prefix.data)
```

should do it

Reply:
Thanks @scheibelp! That fixed the problem.